source/opt/decoration_manager.cpp

```diff
@@ -260,16 +311,25 @@ void DecorationManager::ForEachDecoration(
 void DecorationManager::CloneDecorations(
     uint32_t from, uint32_t to, std::function<void(ir::Instruction&, bool)> f) {
```

Comment:
Given that the decoration now knows about the context, we might be able to remove the call back `f`.  Just have the context analyze the new instructions.

Reply:
I’ll do that.